utils/merge_syft_sbom.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+import json
+import sys
+
+from typing import Any, Callable
+from urllib.parse import urlsplit, quote_plus
+
+
+# Functions to detect components that are reported differently by Syft
+def _is_syft_local_golang_component(component: dict) -> bool:
+    """
+    Check if a Syft Golang reported component is a local replacement.
+
+    Local replacements are reported in a very different way by Cachi2,
+    which is why the same reports by Syft should be removed.
+    """
+    return (
+        component.get("purl", "").startswith("pkg:golang")
+        and component.get("name", "").startswith(".")
+        or component.get("version", "") == "(devel)"
+    )
+
+
+def _is_cachi2_non_pypi_pip_dependency(component: dict) -> bool:
+    """
+    Check if Cachi2 Pip component was fetched a VCS or HTTPS location.
+
+    Cachi2 reports non-pypi components in a different way from Syft, so
+    the reports from Syft need to be removed.
+
+    Unfortunately, there's no way to determine which components are non-Pypi
+    by looking at the Syft report alone. This function is meant to create a
+    list of non-Pypi components from Cachi2's SBOM, then remove the ones reported
+    by Syft in the merged SBOM.
+    """
+    return (
+        component.get("purl", "").startswith("pkg:pypi")
+        and "version" not in component
+    )
+
+
+# Unique key functions
+def _unique_key_cachi2(component: dict) -> str:
+    """Create a unique key from Cachi2 reported components."""
+    url = urlsplit(component["purl"])
+    # omit the query part, since Cachi2 sets it and Syft does not
+    return url.scheme + ":" + url.path
```

Comment:
This also omits the fragment (subpath), that's probably intentional too?

Reply:
It is intentional, since Syft does not set any qualifiers or subpaths (for Golang and Pip, at least)